.eslintrc.js

```diff
@@ -148,6 +148,10 @@ module.exports = {
         ),
       },
     ],
+    '@typescript-eslint/no-unused-vars': [
+      'error',
+      { vars: 'all', args: 'after-used', ignoreRestSiblings: true, argsIgnorePattern: '^_' },
+    ],
```

Comment:
This change extends the eslint-config-airbnb-typescript [rule](https://github.com/airbnb/javascript/blob/6499695ac11c4640ed0f77f8865a1adcb32d3239/packages/eslint-config-airbnb-base/rules/variables.js#L54) previously in action to add an `argsIgnorePattern` for allowing `_` based unused variables ignored by eslint parser.

CC @mui/x

Reply:
:+1: feel free to open a PR to add this to the Core config.